OctoKit/OCTGenericHook.h

```
@@ -0,0 +1,18 @@
+//
+//  OCTGenericHook.h
+//  OctoKit
+//
+//  Created by Benjamin Dobell on 3/8/16.
+//  Copyright (c) 2016 GitHub. All rights reserved.
+//
+
+#import "OCTHook.h"
+
+// A generic representation of a Github hook, used when we encounter hooks that
+// we offer no special case handling for. This class provides bare metal access
+// to config details of any Github hook.
```

This class is perhaps more important than it seems. There exists a problem with other class clusters in this project, for instance events.

At the moment, if a new event were encountered that Octokit does not know about, then transforming the JSON into objects will fail within Mantle. Namely if implementations of `classForParsingJSONDictionary` return nil, then Mantle will throw an error rather than just ignoring the object and moving on. The logic in Mantle is conservative, so I don't really take issue with it. But it does mean that when we're using class clusters we need to ensure that the branch condition is all inclusive, otherwise the API will break. Presently Octokit is not being all inclusive in its class cluster branch conditions e.g. for events, we're looking up based on type, but if we encounter an unknown type there is no fall-back class - so an error will be thrown. This will break deployed projects running in the wild.